src/app/shared/services/ssh-keypair.service.ts

```diff
@@ -23,6 +23,14 @@ export class SSHKeyPairService extends BaseBackendCachedService<SSHKeyPair> {
     super();
   }
 
+  public getByName(name: string): Observable<SSHKeyPair> {
+    return this.getList().map(sshKeyPairs => {
+      return sshKeyPairs.find(sshKeyPair => {
```

Comment:
you can pass a keypair name to the request in listSSHKeyPairs
https://cloudstack.apache.org/api/apidocs-4.9/apis/listSSHKeyPairs.html